mods/bones/init.lua

```diff
@@ -182,41 +196,107 @@ local drop = function(pos, itemstack)
 	end
 end
 
-local player_inventory_lists = { "main", "craft" }
-bones.player_inventory_lists = player_inventory_lists
+bones.player_inventory_lists = { "main", "craft" }
+
+local collect_items_callbacks = {}
+
+function bones.register_collect_items(func)
+	table.insert(collect_items_callbacks, func)
+end
+
+bones.register_collect_items(function(player)
+	local items = {}
+	local player_inv = player:get_inventory()
+	for _, list_name in ipairs(bones.player_inventory_lists) do
+		table.insert_all(items, player_inv:get_list(list_name) or {})
+		player_inv:set_list(list_name, {})
+	end
+	return items
+end)
 
-local function is_all_empty(player_inv)
-	for _, list_name in ipairs(player_inventory_lists) do
-		if not player_inv:is_empty(list_name) then
-			return false
+local function collect_items(player)
+	local items = {}
+	for _, cb in ipairs(collect_items_callbacks) do
+		table.insert_all(items, cb(player))
+	end
+	return items
+end
+
+-- Try to find the closest space near the player to place bones
+local function find_bones_pos(player)
+	local rounded_player_pos = vector.round(player:get_pos())
+	local bones_pos
+	if may_replace(rounded_player_pos, player) then
+		bones_pos = rounded_player_pos
+	else
+		bones_pos = minetest.find_node_near(rounded_player_pos, 1, {"air"})
+	end
+	if minetest.is_protected(bones_pos, player:get_player_name()) then
```

Comment:
This indirectly reverts 110c23560 (the conflicting commit).

When air was found, there should be no protection check.

Reply:
> This indirectly reverts [110c235](https://github.com/minetest/minetest_game/commit/110c23560edf5ddc7ed3ffd0d0c288a96e0343e8) (the conflicting commit).
> 
> When air was found, there should be no protection check.

may_replace also check for protections, that might mean duplicate code. I think the if in 234 is safe to be deleted altogether then